feature_engine/creation/custom_functions.py

```diff
@@ -0,0 +1,10 @@
+"""
+CustomFunctions is a wrapper, which allows MathFunctions to detect if a
+custom function should be processed via pandas.agg() or numpy.apply_over_axes()
+"""
+
+
+class CustomFunctions:
```

Comment:
Do we need this class? it seems to be implementing a very simple functionality. Can this not be a one liner in the main MathFeatures class? 

Besides that, the class name does not immediately tell what the class is doing, we'd need docstrings and typehints.